### PR TITLE
At4 show current building

### PR DESCRIPTION
**Pull request by nicolas-zampieri**
## What does this PR do?
This PR resolves the E2E test failures in outdoor_directions.test.ts on iOS by addressing visibility and interaction issues with map elements and search results.

Key Fixes:

Building Selection (Map): Introduced hidden "shadow markers" (60x60 transparent views) within BuildingPolygon.tsx to ensure building outlines are reliably tappable on iOS, bypassing an issue where polygons reported zero height to Detox.
Search Interaction (Path B):
Improved the BuildingSelector search dropdown by using a bare Text component for results. This prevents touch interception by intermediate containers.
Added keepResultsAfterBlur={true} to prevent the dropdown from disappearing when the keyboard is dismissed.
Simplified the test selection by using a stable testID="search-result-item" and tapping the first result by index.
Test Stability:
Switched Path A to use robust coordinate-based map taps for Hall and CJ buildings.
Added keyboard dismissal and coordinate offsets to search result taps to ensure visibility and accuracy.
Increased waitFor timeouts to 15-20s to account for simulator lag and API response times.

## How to test
- [ ]  Unit tests: npm test mobile/__tests__/MapScreen.test.tsx (all 47 passing)
- [ ]  E2E Tests: npx detox test -c ios.sim.release e2e/outdoor_directions.test.ts

## Review checklist
- [ ] code is readable
- [ ] tests updated/added
- [ ] edge cases handled

### CODE REVIEW

**Comment by nicolas-zampieri** [comment]
This is what added select button looks like now

File: mobile/e2e/outdoor_directions.test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { device, element, by, expect, waitFor } from 'detox';
+
+describe('Outdoor Directions E2E Test', () => {
+
+    it('Path A: should select start and destination via map clicks', async () => {
+        await device.disableSynchronization();
+
+        //Force start at Hall building
+        const hallCoords = { latitude: 45.497170, longitude: -73.579000 };
```
Comment (grylajoie) [inline mobile/e2e/outdoor_directions.test.ts]: This is the problem with SonarQube. there are several times this const declared in different files. If you can, make it so it is a const reusable in each file (Building Coordinates declared in data/building.js maybe)

File: mobile/e2e/outdoor_directions.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        await expect(element(by.id('destination-marker'))).toExist();
+        await element(by.id('map')).atIndex(0).pinch(0.98, 'fast', 0);
+        await new Promise(resolve => setTimeout(resolve, 5000));
+    });
+
+    it('Path B: should select start via Current Location and destination via search', async () => {
+        await device.disableSynchronization();
+
+        // Spoof location to Hall Building for "Current Location"
+        const hallCoords = { latitude: 45.497170, longitude: -73.579000 };
```
Comment (grylajoie) [inline mobile/e2e/outdoor_directions.test.ts]: here too

File: mobile/e2e/location_building_highlight.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        // Map should eventually settle and show the Hall Building as highlighted
+        // The polygon testID format: building-polygon-{id}-highlighted when user is inside.
+        // The id for Hall Building in buildings.js is 'Hall Building'
+        await waitFor(element(by.id('building-polygon-Hall Building-highlighted'))).toExist().withTimeout(15000);
+    });
+
+    it('Scenario 3: Moving to a different building updates the highlight correctly', async () => {
+        await device.disableSynchronization();
+        // Relaunch the app with EV Building coordinates
+        const EVCoords = { latitude: 45.495622, longitude: -73.578121 }; // Inside EV Building
```
Comment (grylajoie) [inline mobile/e2e/location_building_highlight.test.ts]: Only used once but should be a Constant

**Comment by nicolas-zampieri** [comment]
Problem with codecov 2 tests failing regarding googleOAuth:
Screenshot 2026-03-03 at 15 15 16

File: mobile/src/hooks/useGoogleCalendar.ts [hunk 4]
```diff
@@ -57,12 +57,16 @@ export function useGoogleCalendar() {
     native: 'com.concordia.hotfixinprod:/oauth2redirect',
   });
 
-  const [, response, promptAsync] = Google.useAuthRequest({
```
Comment (grylajoie) [inline mobile/src/hooks/useGoogleCalendar.ts]: Here you changed how the google auth works, it s no longer using Google.useAuthRequest()

**Review by grylajoie — CHANGES_REQUESTED** [review]
Google Auth was changed
